bzt/modules/shellexec.py

```diff
@@ -199,10 +201,10 @@ def shutdown(self):
         """
         self.check()
 
-        if self.ret_code is None:
+        if self.process and self.ret_code is None:
             self.log.info("Background task was not completed, shutting it down: %s", self)
             shutdown_process(self.process, self.log)
-        self.process = None
+            self.process = None
```

Comment:
Why put this under `if` ? Why leave chance that there will be process set?

Reply:
I see no such chance. We have two correct points to reset self.process:
1. in check() after communicate() - for correctly finished tasks.
2. after forced shutdown_process (if #1 didn't work)